npm/test/upload-form-files.test.ts

```diff
@@ -0,0 +1,565 @@
+    document.querySelector("#file-upload")
+  dropzone: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__dropzone"
+  )
+  uploadForm: HTMLFormElement | null =
+    document.querySelector("#file-upload-form")
+  fileRetriever: HTMLInputElement | null =
+    document.querySelector("#file-selection")
+
+  fileRetrievalSuccessMessage: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__success"
+  )
+
+  fileNameElement: HTMLElement | null = document.querySelector("#file-name")
+
+  warningMessages: {
+    [s: string]: HTMLElement | null
+  } = {
+    incorrectItemSelectedMessage: document.querySelector(
+      "#item-selection-failure"
+    ),
+    submissionWithoutSelectionMessage: document.querySelector(
+      "#nothing-selected-submission-message"
+    )
+  }
+
+  warningMessagesText: {
+    [s: string]: HTMLElement | null
+  } = {
+    incorrectItemSelectedMessageText: document.querySelector(
+      "#non-file-selected-message-text"
+    )
+  }
+
+  hiddenInputButton: HTMLElement | null =
+    document.querySelector("#file-selection")
+
+  submitButton: HTMLElement | null = document.querySelector(
+    "#start-upload-button"
+  )
+
+  fileUploader = this.setUpFileUploader()
+
+  form = new UploadForm(
+    true,
+    this.uploadForm!,
+    this.fileRetriever!,
+    this.dropzone!,
+    this.setUpFileUploader
+  )
+
+  uploadingRecordsSection = document.querySelector("#upload-progress")
+}
+
+test("clicking the submit button, without selecting a file, doesn't reveal the progress bar", async () => {
+  const mockDom = new MockDom()
+
+  const submitEvent = mockDom.createSubmitEvent()
+  await mockDom.form.handleFormSubmission(submitEvent)
+
+  expect(mockDom.uploadingRecordsSection).toHaveAttribute("hidden")
+})
+
+test("clicking the submit button, without selecting a file, doesn't disable the buttons on the page", async () => {
+  const mockDom = new MockDom()
+
+  const submitEvent = mockDom.createSubmitEvent()
+  await mockDom.form.handleFormSubmission(submitEvent)
+
+  expect(mockDom.submitButton).not.toHaveAttribute("disabled", "true")
+  expect(mockDom.hiddenInputButton).not.toHaveAttribute("disabled", "true")
+})
+
+test("clicking the submit button, without selecting a file, displays a warning message to the user", async () => {
+  const mockDom = new MockDom()
+
+  const submitEvent = mockDom.createSubmitEvent()
+  await mockDom.form.handleFormSubmission(submitEvent)
+
+  expect(
+    mockDom.warningMessages.submissionWithoutSelectionMessage
+  ).not.toHaveAttribute("hidden", "true")
+
+  expect(mockDom.warningMessages.incorrectItemSelectedMessage).toHaveAttribute(
+    "hidden",
+    "true"
+  )
+  expect(mockDom.fileRetrievalSuccessMessage).toHaveAttribute("hidden", "true")
+})
+
+// docx test and wrong number of files test
+// add file name to page
+
+test("input button updates the page with correct number of files if only 1 file has been selected", () => {
```

Comment:
Can a user select more than one file using the browse? If so, should there be a test for this scenario?

Reply:
They shouldn't be able to as, I've now removed the ability to do that yesterday. I'll change it to 

`input button updates the page with the file that has been selected`

..to reflect that.